Closes #341, #334 Speed up MDV2 Dataset Processing

emtwo opened the pull request:
This PR includes:
* Removing deduping
* Switching scalars to use telemetry_stable
* Adding an intermediate step - `clients_scalar_aggregates` and `clients_histogram_aggregates`
* Fixing tests

In terms of the added intermediate step - I was trying to avoid having this for as long as possible, but as more probes were added and processing got slower, this was one method of speeding things up.

fbertsch commented on sql/telemetry/client_histogram_probe_counts_v1/query.sql:
```diff
@@ -177,89 +127,7 @@ RETURNS ARRAY<STRUCT<key STRING, value FLOAT64>> AS (
-  WHERE num_buckets > 0),
-
-clients_aggregates AS
+WITH dedupe_sum AS
```
nit: This CTE name is a bit confusing

fbertsch commented on sql/telemetry/clients_daily_keyed_boolean_aggregates_v1/query.sql:
```diff
@@ -1,38 +1,23 @@
+        normalized_channel AS channel
+    FROM `moz-fx-data-shared-prod.telemetry_stable.main_v4`
+    WHERE DATE(submission_timestamp) = @submission_date
+    AND normalized_channel in (
```
nit: This doesn't match the SQL style guide (indent the `AND`)

fbertsch commented on sql/telemetry/clients_histogram_aggregates_v1/query.sql:
```diff
@@ -0,0 +1,165 @@
+    WHERE DATE(submission_timestamp) > DATE_SUB(CURRENT_DATE(), INTERVAL 90 DAY)
+    AND normalized_channel IN ("nightly", "beta", "release")
+    GROUP BY 1, 2
+    HAVING COUNT(*) > 1000
```
Consider using `COUNT(DISTINCT client_id)` here instead, with e.g. 2k clients. A thousand pings could present a problem at some point.

fbertsch commented on sql/telemetry/clients_scalar_aggregates_v1/query.sql:
```diff
@@ -0,0 +1,161 @@
+    metric_type,
+    key,
+    agg_type,
+    CASE agg_type
```
Are we missing `count`? That's how we'll get `avg`: `sum / count`

emtwo replied:
good catch! I'll update this. Thank you

emtwo commented:
@fbertsch I've updated the PR addressing your comments. The dry-run seems to be failing and wanting certain tables to be available in `moz-fx-data-shared-prod` but I don't have permissions to add them there. Could either request to add them or hardcode it to use `moz-fx-data-derived-datasets` for now. Or I'm missing some config param entirely? Let me know what you think
